File: transaction/src/main/java/br/com/accenture/wallet/transaction/service/ResourceNotFoundException.java
package br.com.accenture.wallet.transaction.service;

public class ResourceNotFoundException extends RuntimeException {

    public ResourceNotFoundException(String entity) {
        super(String.format("Resource '%s' cannot be found.", entity));
    }

    public ResourceNotFoundException(String entity, String id) {
        super(String.format("Resource '%s' cannot be found with '%s'.", entity, id));
    }

}

File: customer/src/main/kotlin/br/com/accenture/wallet/customer/service/ResourceNotFoundException.kt
package br.com.accenture.wallet.customer.service

class ResourceNotFoundException : RuntimeException {

    constructor(entity: String) : super(String.format("Resource '%s' cannot be found.", entity))

    constructor(entity: String, id: String) : super(String.format("Resource '%s' cannot be found with '%s'.", entity, id))

}
